app/src/main/java/com/android/sample/ui/camera/CameraScanCodeBarScreen.kt

```diff
@@ -147,93 +183,142 @@ fun IngredientOverlay(
+        contentScale = ContentScale.Crop,
+    )
+  }
+}
+
+@Composable
+private fun IngredientNameAndBrand(ingredient: Ingredient) {
+  Text(
+      text = ingredient.name,
+      style = MaterialTheme.typography.titleSmall,
+      modifier =
+          Modifier.padding(
+              vertical =
+                  C.Dimension.CameraScanCodeBarScreen.INGREDIENT_DISPLAY_TEXT_NAME_PADDING_V.dp,
+              horizontal =
+                  C.Dimension.CameraScanCodeBarScreen.INGREDIENT_DISPLAY_TEXT_NAME_PADDING_H.dp))
+  Text(
+      text = ingredient.brands ?: "",
+      style = MaterialTheme.typography.bodySmall,
+      modifier =
+          Modifier.padding(
+              vertical =
+                  C.Dimension.CameraScanCodeBarScreen.INGREDIENT_DISPLAY_TEXT_BRAND_PADDING_V.dp,
+              horizontal =
+                  C.Dimension.CameraScanCodeBarScreen.INGREDIENT_DISPLAY_TEXT_BRAND_PADDING_H.dp))
+}
+
+@Composable
+private fun IngredientSelectButton(
+    ingredient: Ingredient,
+    searchIngredientViewModel: SearchIngredientViewModel,
+    navigationActions: NavigationActions
+) {
+  Button(
+      onClick = {
+        searchIngredientViewModel.addIngredient(ingredient)
+        navigationActions.navigateTo(Screen.CREATE_RECIPE_LIST_INGREDIENTS)
+      },
+      modifier =
+          Modifier.padding(
+              vertical =
+                  C.Dimension.CameraScanCodeBarScreen.INGREDIENT_DISPLAY_TEXT_BUTTON_PADDING_V.dp,
+              horizontal =
+                  C.Dimension.CameraScanCodeBarScreen.INGREDIENT_DISPLAY_TEXT_BUTTON_PADDING_H
+                      .dp)) {
+        Text(
+            text = stringResource(R.string.add_to_fridge),
+            style = MaterialTheme.typography.bodySmall,
+            color = MaterialTheme.colorScheme.onPrimary,
+        )
+      }
+}
+
+@Composable
+private fun IngredientNotFoundDisplay() {
+  Log.i("IngredientDisplay", "IngredientNotFoundDisplay")
+  Box(
+      modifier =
+          Modifier.fillMaxSize()
+              .padding(C.Dimension.CameraScanCodeBarScreen.INGREDIENT_DISPLAY_PADDING.dp),
+      contentAlignment = Alignment.Center) {
+        Text(
+            text = stringResource(R.string.ingredient_not_found),
+            style = MaterialTheme.typography.bodyLarge,
+            modifier =
+                Modifier.padding(
+                    vertical =
+                        C.Dimension.CameraScanCodeBarScreen.INGREDIENT_DISPLAY_TEXT_NAME_PADDING_V
+                            .dp,
+                    horizontal =
+                        C.Dimension.CameraScanCodeBarScreen.INGREDIENT_DISPLAY_TEXT_NAME_PADDING_H
+                            .dp))
+      }
+}
+
+@Preview
+@Composable
+private fun IngredientBeingFetchedDisplay() {
+  Box(
+      modifier =
+          Modifier.fillMaxSize()
+              .padding(C.Dimension.CameraScanCodeBarScreen.INGREDIENT_DISPLAY_PADDING.dp),
+      contentAlignment = Alignment.Center) {
+        LoadingCook()
+      }
+}
```

Comment:
I really like the idea! It would be even better if the little Cook icon rotated to symbolize the loading process. I recall that @AdriB02 implemented something similar on another screen, do you happen to know if it would be difficult to implement here?

Reply:
It does rotate! It's just that the ingredient is usually found before you see it. I reuse Adrien's function for it.

Reply:
Nice!